From ea61a67a3da2be6efb1678b81dc25fca4581a563 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:18 -0400
Subject: [tools] turbostat: Don't attempt to printf an off_t with %zx

Message-id: <1409057382-26432-8-git-send-email-prarit@redhat.com>
Patchwork-id: 89085
O-Subject: [RHEL7.1 PATCH BZ 1133893 07/31] turbostat: Don't attempt to printf an off_t with %zx
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 2e9c6bc7fb6ffc32d83bc133e4a7389125e8eb0a
Author: Josh Triplett <josh@joshtriplett.org>
Date:   Tue Aug 20 17:20:13 2013 -0700

    turbostat: Don't attempt to printf an off_t with %zx

    turbostat uses the format %zx to print an off_t.  However, %zx wants a
    size_t, not an off_t.  On 32-bit targets, those refer to different
    types, potentially even with different sizes.  Use %llx and a cast
    instead, since printf does not have a length modifier for off_t.

    Without this patch, when compiling for a 32-bit target:

    turbostat.c: In function 'get_msr':
    turbostat.c:231:3: warning: format '%zx' expects argument of type 'size_t', but argument 4 has type 'off_t' [-Wformat]

    Signed-off-by: Josh Triplett <josh@joshtriplett.org>
    Signed-off-by: Len Brown <len.brown@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Dean Nelson <dnelson@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index 013d1c4..14de623 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -253,7 +253,7 @@ int get_msr(int cpu, off_t offset, unsigned long long *msr)
  close(fd);
 
  if (retval != sizeof *msr) {
-  fprintf(stderr, "%s offset 0x%zx read failed\n", pathname, offset);
+  fprintf(stderr, "%s offset 0x%llx read failed\n", pathname, (unsigned long long)offset);
   return -1;
  }
 
-- 
1.7.1